src/commonMain/kotlin/exchange.dydx.abacus/processor/wallet/account/FillsProcessor.kt

```diff
@@ -1,15 +1,34 @@
 package exchange.dydx.abacus.processor.wallet.account
 
+import exchange.dydx.abacus.output.input.MarginMode
 import exchange.dydx.abacus.processor.base.BaseProcessor
 import exchange.dydx.abacus.processor.base.mergeWithIds
 import exchange.dydx.abacus.protocols.ParserProtocol
+import exchange.dydx.abacus.utils.NUM_PARENT_SUBACCOUNTS
+import exchange.dydx.abacus.utils.safeSet
 
 internal class FillsProcessor(parser: ParserProtocol) : BaseProcessor(parser) {
     private val itemProcessor = FillProcessor(parser = parser)
 
-    override fun received(existing: List<Any>?, payload: List<Any>): List<Any>? {
+    fun received(existing: List<Any>?, payload: List<Any>, subaccountNumber: Int?): List<Any>? {
         val new = payload.mapNotNull { eachPayload ->
-            parser.asNativeMap(eachPayload)?.let { eachPayloadData -> itemProcessor.received(null, eachPayloadData) }
+            parser.asNativeMap(eachPayload)?.let { eachPayloadData ->
+                val modified = eachPayloadData.toMutableMap()
```

Comment:
I think it'd be cleaner to do this in the FillProcessor (i.e., passing the subaccountNumber over), since technically we should modify the output instead of the input payload.